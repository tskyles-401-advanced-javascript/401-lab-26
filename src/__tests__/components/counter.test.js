import Enzyme, { mount, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";
import renderer from "react-test-renderer";
import Counter from "../../components/counter.js";

Enzyme.configure({ adapter: new Adapter() });

describe("<Counter />", () => {
  it("is alive at app start", () => {
    let app = shallow(<Counter />);
    expect(app.find("form").exists()).toBeTruthy();
  });

  it("increases and decreases count when buttons are clicked", () => {
    let app = mount(<Counter />);
    let buttonUp = app.find(".positive");
    let buttonDown = app.find(".negative");
    buttonUp.simulate("click");
    expect(app.state("count")).toBe(1);
    buttonDown.simulate("click");
    expect(app.state("count")).toBe(0);
  });

  it('renders right', () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  })
});
