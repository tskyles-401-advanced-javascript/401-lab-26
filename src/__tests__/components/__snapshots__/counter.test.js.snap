// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders right 1`] = `
<div>
  <form>
    <h4>
      Choose a starting number
    </h4>
    <input
      onChange={[Function]}
      type="number"
    />
  </form>
  <h4>
    The number is
  </h4>
  <h2
    id="counter"
    style={
      Object {
        "color": "black",
      }
    }
  >
    0
  </h2>
  <button
    className="button positive"
    onClick={[Function]}
  >
    Increase
  </button>
  <button
    className="button negative"
    onClick={[Function]}
  >
    Decrease
  </button>
</div>
`;
